mindsdb/libs/phases/stats_generator/stats_generator.py

```diff
@@ -223,32 +181,202 @@ def getParamsAsDictionary(self, params):
         }
         return ret
 
+    def value_distribution_score(self, stats, columns, col_name):
+        bucket_probabilities = {}
+        pair = stats[col_name]['histogram']
+        total_vals = sum(pair['y'])
+        for i in range(len(pair['x'])):
+            bucket_probabilities[pair['x'][i]] = pair['y'][i]/total_vals
+
+        probabilities = list(bucket_probabilities.values())
+        mean_max_distribution_diff = 1 - np.mean(probabilities)/max(probabilities)
+        max_probability = max(probabilities)
+
+        value_distribution_score =  (mean_max_distribution_diff - max_probability)/100
+
+        data = {
+            'bucket_probabilities': bucket_probabilities
+            ,'value_distribution_score': value_distribution_score
+        }
+
+        return data
+
+
+    def duplicates_score(self, stats, columns, col_name):
+        duplicates = len(columns[col_name]) - len(set(columns[col_name]))
+        data = {
+            'duplicates': duplicates
+            ,'duplicates_percentage': duplicates*100/len(columns[col_name])
+        }
+
+        if stats[col_name][KEYS.DATA_TYPE] != DATA_TYPES.CATEGORICAL and stats[col_name][KEYS.DATA_TYPE] != DATA_TYPES.DATE:
+            data['duplicate_score'] = data['duplicates_percentage']/100
+        else:
+            data['duplicate_score'] = 0
+
+        return data
+
+    def empty_cells_score(self, stats, columns, col_name):
```

Comment:
since we are moving it all to underscores, in order to differentiate methods from variables, methods should be verbs, so calculate_empty_cells_score

Reply:
again, will do

Reply:
Fixed in #9880798afad27509d19568c36cdd9a5732c3aeb8